src/main/java/cuchaz/enigma/gui/GuiController.java

```diff
@@ -398,14 +413,15 @@ private void loadClass(ClassEntry classEntry, Runnable callback) {
 
 	private DecompiledClassSource decompileSource(ClassEntry targetClass, SourceProvider sourceProvider) {
 		try {
-			CompilationUnit sourceTree = sourceProvider.getSources(targetClass.getFullName());
+			CompilationUnit sourceTree = (CompilationUnit) sourceProvider.getSources(targetClass.getFullName()).clone();
```

Comment:
Doesn't this always return `CompilationUnit`?

Reply:
The clone method returns an `AstNode`.

Reply:
Oh, oops; didn't see that! 😄